### PR TITLE
fix: dont fetch a route for wrap transactions

### PR DESCRIPTION
**Pull request by JFrankfurt**

### CODE REVIEW

**Review by zzmp — COMMENTED** [review]
Does this gain us anything in the UI, or is it just an optimization to reduce backend calls? I'm not sure if it's worthwhile to add this complexity given that the widget is slated to be swapped in soon.

**Comment by JFrankfurt** [comment]
This fixes a UI bug: https://www.notion.so/uniswaplabs/Token-price-quotes-impact-shouldn-t-render-for-wrapping-57e46736ab9d44a981a7d1a0332123f8?pvs=4#ce847178e1754f76a85fe73b54c45aaa

File: src/hooks/useBestTrade.ts [hunk 1]
```diff
@@ -32,16 +36,28 @@ export function useBestTrade(
     200
   )
 
+  const wethContract = useWETHContract()
+  const isAWrapTransaction = useMemo(() => {
+    if (!wethContract || !chainId || !amountSpecified || !debouncedOtherCurrency) return undefined
```
Comment (zzmp) [inline src/hooks/useBestTrade.ts:41]: This should only return a boolean, and then you shouldn't need to cast `shouldGetTrade` to a boolean.

```suggestion
    if (!wethContract || !chainId || !amountSpecified || !debouncedOtherCurrency) return false
```

Comment (zzmp) [inline src/hooks/useBestTrade.ts:41]: Is it necessary to check if a `wethContract` exists, or is it enough to just check that the currency pair is native/weth?

File: src/hooks/useBestTrade.ts [hunk 2]
```diff
@@ -32,16 +36,28 @@ export function useBestTrade(
...
+    const weth = chainId ? WRAPPED_NATIVE_CURRENCY[chainId] : undefined
+    return (
+      (amountSpecified.currency.isNative && weth?.equals(debouncedOtherCurrency)) ||
+      (debouncedOtherCurrency.isNative && weth?.equals(amountSpecified.currency))
+    )
+  }, [amountSpecified, chainId, debouncedOtherCurrency, wethContract])
+
+  const shouldGetTrade = Boolean(!isAWrapTransaction && isWindowVisible)
```
Comment (zzmp) [inline src/hooks/useBestTrade.ts:49]:
```suggestion
  const shouldGetTrade = !isAWrapTransaction && isWindowVisible
```

**Review by tinaszheng — APPROVED** [review]

**Pull request merged**